code/io/joy-sdl.cpp

```diff
@@ -801,7 +856,7 @@ int joy_down(int btn)
 	else if (btn >= JOY_NUM_BUTTONS)
 	{
 		// Is hat
-		return (current->getHatPosition(0) == btn) ? 1 : 0;
+		return current->isHatDown(0, btn);
```

Comment:
By changing this from a comparison with the enum to an index without reverting commit 9f12e1f749d9bd6ba51166ab745e6b14896d6daa, this breaks hats again.